syntax = "proto3";

import "google/protobuf/timestamp.proto";

package api.treeserve.v1;
option go_package = "/pkg/pb/treeserve;treeserve";

service DeploymentService {
  rpc Deploy(DeployRequest) returns (DeployResponse);
  rpc Info(InfoRequest) returns (InfoResponse);
  rpc Predict(PredictRequest) returns (PredictResponse);
}

message DeployRequest {
  string model_name = 1;
}
message DeployResponse {
  string model_name = 1;
}

message InfoRequest{
  string model_name = 1;
}

message InfoResponse {
  string model_name = 1;
  string model_version = 2;
  google.protobuf.Timestamp created_ts = 3;
}

message PredictRequest {
  string model_name = 1;
  string input_data = 2;
}

message PredictResponse {
  string model_name = 1;
 repeated double predictions = 3;
}

